### PR TITLE
Fixed: user status offline must be updated too

### LINKED ISSUE #214: User status not updated properly
Your Rocket.Chat version: demo

Hi @rafaelks ,

the user status is not synced properly, can you recheck this?

Here are some screenshots: (made on the same time)

(User status on my iPhone, pretty much users online on the demo :-D)

When I look into my electron client.. and .. surprising :( not that much users are online:

bildschirmfoto 2017-01-08 um 22 03 28

### PR DESCRIPTION
**Pull request by rafaelks**
@RocketChat/iOS

Closes #214.

### CODE REVIEW

**Comment by rafaelks** [comment]
@laggedHero @rakaramos Can you guys review the PR? I've a few ideas to start changing the architecture to "clean", but I think we should not do it with this PR. For this one (since is a bug fix) let's focus on the bug fix only.

File: Rocket.Chat/Managers/Socket/Response/SocketHandlers.swift [hunk 1]
```diff
@@ -78,8 +78,19 @@ extension SocketManager {
...
+                }
+
+                if msg == .Removed {
+                    let user = Realm.getOrCreate(User.self, primaryKey: identifier, values: fields)
+
+                    Realm.execute({ (realm) in
+                        user.status = .offline
+                        realm.add(user, update: true)
+                    })
+                }
```
Comment (rafaelks) [inline Rocket.Chat/Managers/Socket/Response/SocketHandlers.swift]: @rakaramos @laggedHero First change I'll made is remove the Realm dependency from this file, creating the initial interactions with DB domain to update the objects.

File: Rocket.Chat/Managers/Launcher/UserCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    
+    // This method is suppose to be called when the app starts. We want all
+    // users to be offline, because when we subscribe to the API that returns
+    // the active users, it will only return users that aren't offline.
+    private func turnAllUsersOffline() {
+        // FIXME: Remove the Realm dependency.
+        Realm.execute { (realm) in
+            let users = realm.objects(User.self)
+            users.setValue("offline", forKey: "privateStatus")
+        }
```
Comment (rafaelks) [inline Rocket.Chat/Managers/Launcher/UserCoordinator.swift]: Once the interactors with DB are ready, we can remove this dependency too.

**Comment by laggedHero** [comment]
I think it's fine. Moving the dependency on Realm "out" will take time. Let's make sure to not slow down too much because of that.

:)

**Pull request merged**